### PR TITLE
fix: adjust node_modules caching for io preview builds

### PR DESCRIPTION
**Pull request by BRKalow**
## "Ready for Review" checklist

- [ ] The Vercel preview link has been added to this PR's description
- [ ] The Asana task has been added to this PR's description
- [ ] This PR's link has been added to the "📐 Pull Request" field on the Asana task
- [ ] This Vercel preview link has been added to the "🔍 Preview" field on the Asana task
- [ ] The description template has been filled in below

---

## Relevant links

- [Preview link](url) 🔎
- [Asana task](url) 🎟️

## What

- Adjusting how we are caching `node_modules` for deploy preview builds in other projects.
- Adjust logic for showing the product switcher so that it doesn't show in deploy previews in other projects.
- Remove unneeded configuration option in `next.config.js`

## Why

- Making some adjustments based on errors encountered during another round of our deploy preview validation.
- Removed `outputFileTracing: false` because Next rolled out a fix in a patch release of v12 after we reported an issue that required us to disable it.

## Testing

- [ ] Observe this build is _green_: https://vercel.com/hashicorp/waypoint/H1WgEwSJC5KQNMEoNC2CRGyLPfcJ

## Anything else?

It seems like we're also getting some nice build-time improvement here, too. I'm guessing it's due to the removal of `outputFileTracing: false`. h/t to @dstaley for seeing it!

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1201931300013821

### CODE REVIEW

File: next.config.js [hunk 1]
```diff
@@ -120,7 +120,6 @@ module.exports = withSwingset({
         },
       ],
     },
-    outputFileTracing: false,
```
Comment (BRKalow) [inline next.config.js:123]: This was turned off due to a bug, which was fixed in a v12 patch release shortly after we reported it.

File: scripts/content-repo-preview/build.ts [hunk 2]
```diff
@@ -101,14 +96,8 @@ async function main() {
   /** Build */
   execFileSync('npm', ['run', 'build'], { stdio: 'inherit' })
 
-  // Copy the .next folder to the top-level so the site can be deployed and we can take advantage of the cache
-  // We are using cp here because the Vercel build is still running in the context of the nested website-preview directory,
-  // and so it needs the .next folder available to complete the deployment
-  execFileSync('cp', ['-R', '.next', '../.next'], { stdio: 'inherit' })
```
Comment (BRKalow) [inline scripts/content-repo-preview/build.ts:107]: We're pointing the `outputDirectory` vercel config option to `website-preview/.next` going forward, so we don't need to check for it at the top level like this.

File: scripts/content-repo-preview/build.ts [hunk 3]
```diff
@@ -101,14 +96,8 @@ async function main() {
...
-
   // Put node_modules into .next/cache so we can retrieve them on subsequent builds
-  // We are using mv here because it is more efficient than cp, and node_modules is a big chungus
-  execFileSync('mv', ['node_modules', '../.next/cache/node_modules'], {
```
Comment (BRKalow) [inline scripts/content-repo-preview/build.ts:111]: We can't use `mv` here unfortunately, as the deployment process depends on the node_modules being in the right spot 😅  Using cp is a bit slower, but doing this still provides a nice speedup for builds.

**Review by dstaley — APPROVED** [review]
🚤 🚤 🚤 🚤

**Pull request merged**